### ISSUE TITLE
Gni & Android Hotspot

### ISSUE THREAD

**Issue by DJAlo** [Genymobile/gnirehtet#373]
Hi, I'm trying to make Gni work with Hotspot active on the phone to share Internet via WiFi from phone.
When I set everything up there's no Internet on Hotspot and on console log a lot of dropped packets (Probably everyone that device sends?)
Anyone tried this? Tried on older Redmi Note 3 (Can't connect to Hotspot after turning Gni) and Redmi Note 5 (Hotspot is on, but only errors of packets in console log). Any workaround? :)

**Comment by rom1v** [comment]
Yes, that does not work. #95

**Comment by Himanshu9152704239** [comment]
use every proxy to "bridge USB" Tethering and wifi hotspot
https://play.google.com/store/apps/details?id=com.gorillasoftware.everyproxy&hl=en_IN&gl=US 

http             | toggled On
https           | toggled  On

All IP to your default  gateway (Tethering IP Address)

on other devices connect to your hotspot and use proxy 
which you set up on Every Proxy (example:- IP address: Port [192.168.1.1:8080])

**Comment by shenmo7192** [comment]
> 

Great! @rom1v This may be a solution

**Comment by shenmo7192** [comment]
> use every proxy to "bridge USB" Tethering and wifi hotspot https://play.google.com/store/apps/details?id=com.gorillasoftware.everyproxy&hl=en_IN&gl=US
> 
> http | toggled On https | toggled On
> 
> All IP to your default gateway (Tethering IP Address)
> 
> on other devices connect to your hotspot and use proxy which you set up on Every Proxy (example:- IP address: Port [192.168.1.1:8080])

For super old devices(For me is a HUAWER HONOR 5 Lite), it only support a older proxy app. Called proxy server. Available in apkpure

**Comment by Corpseff** [comment]
> use every proxy to "bridge USB" Tethering and wifi hotspot https://play.google.com/store/apps/details?id=com.gorillasoftware.everyproxy&hl=en_IN&gl=US
> 
> http | toggled On https | toggled On
> 
> All IP to your default gateway (Tethering IP Address)
> 
> on other devices connect to your hotspot and use proxy which you set up on Every Proxy (example:- IP address: Port [192.168.1.1:8080])

yes it has worked for web connections. but can't use with steam link. my phone doesn't see the steam client.

**Comment by zyx486013** [comment]
you can try this:https://github.com/xinruoyusixian/android-reverse-tethering-master
I modify a Xray version of this :https://github.com/zyx486013/android-reverse-tethering-v2-win7x86
Based on my testing, the Xray version supports VPN Hotspot or the built-in SOCKS proxy.
